incubator/node/templates/deployment.yaml

```diff
@@ -0,0 +1,51 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "fullname" . }}
+  labels:
+    app: {{ template "fullname" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: {{ template "fullname" . }}
+    spec:
+      containers:
+      - name: {{ template "fullname" . }}
+        image: "{{ .Values.image }}:{{ .Values.imageTag }}"
+        imagePullPolicy: {{ default "" .Values.imagePullPolicy | quote }}
+        command: [ "sh" ]
+        args: [ "-c" , "git clone {{ .Values.gitRepository }} /app &&  cd /app &&  git checkout {{ .Values.gitBranch}}  &&  npm install &&  npm start" ] 
```

Comment:
Makes sense to me use the gitrepo volume plugin so there's not any need to attach a PV.
I'm gonna update the PR now.